src/net_processing.cpp

```diff
@@ -585,10 +585,10 @@ class PeerManagerImpl final : public PeerManager
      * Reconsider orphan transactions after a parent has been accepted to the mempool.
      *
      * @peer[in]  peer     The peer whose orphan transactions we will reconsider. Generally only one
-     *                     orphan will be reconsidered on each call of this function. This set
-     *                     may be added to if accepting an orphan causes its children to be
+     *                     orphan will be reconsidered on each call of this function. May
```

Comment:
1. I think it should say "Creates more work if", as it's less ambiguous (I actually spent time figuring the difference out...).
2. So now `bool fMoreNodeWork = m_msgproc->ProcessMessages(pnode, flagInterruptMsgProc);` should be renamed to `may_have_more_work` right?

Reply:
I don't think this PR changes whether `fMoreNodeWork` should be renamed -- the bool should have the same properties both before and after this PR.

But... I think its current behaviour is technically not quite right? If we process a TX message in `ProcessMessage` now, we'll set `fMoreWork = true` only when `m_getdata_requests` has entries, but I think we should also be doing it if the peer's orphanage work queue isn't empty, as `ProcessMessage` may have added to it. I think this is technically a bug introduced in #15644 -- there should have been an `if (!pfrom->orphan_work_set.empty()) fMoreWork = true;`  added after the `ProcessMessage()` call, as well as after the `// this maintains the order of responses` comment.

The only result of the bug is that when a parent comes in, processing its orphaned children will be delayed by 100ms rather than occurring essentially immediately.

Reply:
Maybe it would be better to write:

```c++
while (true) {
    bool all_nodes_finished_processing = true;
    for (pnode : nodes) {
        if (m_msgproc->ProcessMessages(pnode)) all_nodes_finished_processing = false;
        m_msgproc->SendMessages(pnode);
    }
    if (all_nodes_finished_processing) wait_for(100ms);
}
```

The idea (as I understand it) is just that `ProcessMessages()` returns true if it does some substantial work and exits early, without checking everything. If that's the case for any peer, then it doesn't wait before looping through all peers again. If `ProcessMessages()` completed for all peers, whether it did work or not, then it is okay to sleep briefly before looping over everything again.

Reply:
I think you are right. Let's try applying these changes and see whether everything makes sense once again.

Part of my confusion is that it's unclear whether `has_more_work` applies to a given peer, or to all peers... That variable rename could help. (`peer_has_more_work` vs `node_has_more_work` or something.)